pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -640,6 +640,8 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
                 if candidate.finish_reason == 'STOP':  # pragma: no cover
                     # Normal completion - skip this chunk
                     continue
+                elif candidate.finish_reason == 'MALFORMED_FUNCTION_CALL':  # pragma: no cover
```

Comment:
Can we add a test for this, based on a real-world response with this finish reason?

Is skipping the chunk the right thing to do? Would any more chunks follow or would this be the last one in the response? Is it valid to just move on and ignore the error? 

If the response before this failure is empty, we retry the request, which I suppose is the goal:

https://github.com/pydantic/pydantic-ai/blob/9b1913ef16f75b3575018a727f36ca25b1470372/pydantic_ai_slim/pydantic_ai/_agent_graph.py#L567-L572